store/wallet.ts

```diff
@@ -22,6 +40,30 @@ export const wallet = createSlice({
   reducers: {
     setStatus: (state, action: PayloadAction<WalletStatus>) => {
       state.status = action.payload
+    },
+    setTokens: (state, action: PayloadAction<AddressToken[]>) => {
+      state.tokens = action.payload
+    },
+    setUtxoBalance: (state, action: PayloadAction<string>) => {
+      state.utxoBalance = action.payload
+    },
+    setAddress: (state, action: PayloadAction<string>) => {
+      state.address = action.payload
     }
   }
 })
+
+const rawTokensSelector = createSelector((state: WalletState) => state.tokens, (tokens) => {
+  if (tokens.some((t) => t.id === '0')) {
+    return tokens
+  } else {
+    return [initialDFI, ...tokens]
+  }
+})
+//* Use as token selector to include UTXO balance
+export const tokensSelector = createSelector([rawTokensSelector, (state: WalletState) => state.utxoBalance], (tokens, utxoBalance) => tokens.map((t) => {
+  if (t.id === '0') {
+    return { ...t, amount: new BigNumber(t.amount).plus(utxoBalance).toFixed() }
+  }
+  return t
+}))
```

Comment:
Hesitant in putting UI logic in the global state as it is only unique to the balances page.

Reply:
@fuxingloh it's not in the global state. It's only in the selector. Also, it will also be used in DEX (whenever you select a token).